eve/workers/openstack-single-node-rhel/terraform/bootstrap.tf

```diff
@@ -0,0 +1,54 @@
+resource "openstack_compute_instance_v2" "bootstrap" {
+  name        = "${local.prefix}-bootstrap"
+  image_name  = var.openstack_image_name
+  flavor_name = var.openstack_flavour_name
+  key_pair    = openstack_compute_keypair_v2.local_ssh_key.name
+
+  security_groups = [
+    openstack_networking_secgroup_v2.bootstrap.name,
+  ]
+
+  dynamic "network" {
+    for_each = [
+      var.openstack_network,
+    ]
+
+    content {
+      name = network.value.name
+    }
+  }
+
+  depends_on = [
+  ]
+
+  connection {
+    host        = self.access_ip_v4
+    type        = "ssh"
+    user        = "cloud-user"
+    private_key = file("~/.ssh/terraform")
+  }
+
+  provisioner "file" {
+    source      = "${path.module}/scripts"
+    destination = "/home/cloud-user/scripts"
+  }
+
+  provisioner "remote-exec" {
+    inline = [
+      "sudo bash scripts/rhsm-register.sh '${var.rhsm_username}' '${var.rhsm_password}'",
+      "sudo bash scripts/bootstrap-config.sh",
+    ]
+  }
+
+  provisioner "remote-exec" {
+     when = "destroy"
+     on_failure = "continue"
```

Comment:
From the RedHat web site, anyway it should not fail, but at least we destroy the VM and release openstack resources.
If it fails, the worker will be destroyed at some point and ssh keys to access VM will be lost, so not sure it's better to have both the VM to destroy and the subscription to remove.

Reply:
I'm not saying it should fail, just wondering what will happen if we fail to remove the subscription and the VM gets destroyed.

Reply:
The `on_failure: "continue"` is more here to avoid the destruction to fail if there is an error when trying to spawn the VM or if the machine is not registered yet (or anymore).

Reply:
OK now with your explanations I'm lost.

What I understand of a Terraform provisioner `on_failure` argument is that it controls how the provisioner's result affects the creation (or deletion, in that case) of the resource, here, the VM.
So, having `on_failure = "continue"` here means, if the "unregister_rhsm" script fails, the VM will still be destroyed.

My question is only: if the script fails, for some unknown reason, and the VM then gets destroyed as expected, do we need some manual action on RedHat side to unregister this now non-existent VM?

Reply:
Yes, if the VM has been registered.

Reply:
OK, will see if we can remember those to cleanup after the fact.